cli/link.ts

```diff
@@ -0,0 +1,65 @@
+import type {ArgumentsCamelCase} from 'yargs';
+import {createCodaClient} from './helpers';
+import {formatEndpoint} from './helpers';
+import {getApiKey} from './config_storage';
+import {getPackId} from './config_storage';
+import {printAndExit} from '../testing/helpers';
+import {promptForInput} from '../testing/helpers';
+import {storePackId} from './config_storage';
+
+// Regular expression that matches coda.io/p/<packId> or <packId>.
+const PackUrlRegex = /^(?:https:\/\/(?:[^/]*)coda.io(?:\:[0-9]+)?\/p\/)?([0-9]+)(:?[^0-9]*)$/;
+
+interface LinkArgs {
+  manifestDir: string;
+  codaApiEndpoint: string;
+  packIdOrUrl: string;
+}
+
+export function parsePackIdOrUrl(packIdOrUrl: string): number | null {
+  const match = packIdOrUrl.match(PackUrlRegex);
+  if (!match) {
+    return null;
+  }
+  const matchedNumber = match[1];
+  return Number(matchedNumber);
+}
+
+export async function handleLink({manifestDir, codaApiEndpoint, packIdOrUrl}: ArgumentsCamelCase<LinkArgs>) {
```

Comment:
Nit, would put this above parsePackIdOrUrl so it's the first function in the file, by convention we try to keep the "public" thing up top so you don't have to scan down too far to find the meat of the file.